src/olympia/addons/models.py

```diff
@@ -541,6 +546,32 @@ def from_upload(cls, upload, platforms, source=None, is_listed=True,
 
         return addon
 
+    @classmethod
+    def resolve_webext_translations(cls, data, upload):
+        """Resolve all possible translations from an add-on.
+
+        This returns a modified `data` dictionary accordingly with proper
+        translations filled in.
+        """
+        if not data.get('is_webextension') or not data.get('default_locale'):
+            # Don't change anything if we don't meet the requirements
+            return data
+
+        fields = ('name', 'homepage', 'summary')
+        messages = extract_translations(upload)
+
+        for field in fields:
+            data[field] = {
+                locale: resolve_i18n_message(
```

Comment:
ah, eh, that comment was weird and wrong. What I should have said was, we already use the correct locale codes in `messages` - see https://github.com/mozilla/addons-server/pull/3012/files/06001951ede3abf01292ebfae7a820d231e204f9#diff-910d9d00817510d6161315516feecfcbR921 for more details. I feel like that should suffice.